### PR TITLE
Rwa 1768 multiple calendars

### PR DESCRIPTION
**Pull request by chrisdavecm**
### JIRA link (if applicable) ###
[RWA-1768](https://tools.hmcts.net/jira/browse/RWA-1768)

### Change description ###
multiple calendars

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by rasikasri — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/PublicHolidaysCollectionTest.java [hunk 1]
```diff
@@ -21,16 +23,30 @@ class PublicHolidaysCollectionTest {
...
+    @Test
+    public void secondCalendarShouldOverrideMainCalendar() {
+        List<String> oneUri = List.of(CALENDAR_URI, "https://raw.githubusercontent.com/hmcts/wa-task-management-api/RWA-1768-calendar-for-test/src/test/resources/override-working-day-calendar.json");
+        Set<LocalDate> oneCalendarResult = publicHolidaysCollection.getPublicHolidays(oneUri);
+        assertThat(oneCalendarResult.contains(LocalDate.of(2022, 12, 26)));
+        assertThat(oneCalendarResult.contains(LocalDate.of(2022, 12, 27)));
+
+        List<String> twoUris = List.of(CALENDAR_URI, "https://raw.githubusercontent.com/hmcts/wa-task-management-api/RWA-1768-calendar-for-test/src/test/resources/override-working-day-calendar.json");
+        Set<LocalDate> fullResult = publicHolidaysCollection.getPublicHolidays(twoUris);
+        assertThat(!oneCalendarResult.contains(LocalDate.of(2022, 12, 26)));
```
Comment (serdaroguzhanakin) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/PublicHolidaysCollectionTest.java:48]: `fullResult` not used and could not understand how `oneCalendarResult` changed :)

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/calendar/BankHolidaysTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+public class BankHolidaysTest {
+
+    @Test
+    void should_create_full_object_and_checkDates() throws IOException {
+        ObjectMapper om = new ObjectMapper();
+        BankHolidays bankHolidays = om.readValue(new URL(DEFAULT_NON_WORKING_CALENDAR), BankHolidays.class);
+        assertTrue(bankHolidays.getDivision().equals("england-and-wales"));
+        assertTrue(bankHolidays.getEvents().size() > 0);
+        assertTrue(bankHolidays.getEvents().get(0).hashCode() != 0);
+        assertTrue(!bankHolidays.getEvents().get(0).equals(null));
```
Comment (serdaroguzhanakin) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/calendar/BankHolidaysTest.java:24]: assertNotNull can be used for this validation?

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/calendar/BankHolidaysTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.gov.hmcts.reform.wataskmanagementapi.services.calendar.DateCalculator.DEFAULT_NON_WORKING_CALENDAR;
+
+public class BankHolidaysTest {
+
+    @Test
+    void should_create_full_object_and_checkDates() throws IOException {
```
Comment (serdaroguzhanakin) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/calendar/BankHolidaysTest.java:18]: method name should be change to this `should_create_full_object_and_check_dates`

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/calendar/BankHolidaysTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        BankHolidays bankHolidays = om.readValue(new URL(DEFAULT_NON_WORKING_CALENDAR), BankHolidays.class);
+        assertTrue(bankHolidays.getDivision().equals("england-and-wales"));
+        assertTrue(bankHolidays.getEvents().size() > 0);
+        assertTrue(bankHolidays.getEvents().get(0).hashCode() != 0);
+        assertTrue(!bankHolidays.getEvents().get(0).equals(null));
+        assertTrue(bankHolidays.getEvents().get(0).equals(bankHolidays.getEvents().get(0)));
+
+        for (BankHolidays.EventDate eventDate: bankHolidays.getEvents()) {
+            assertTrue(isValid(eventDate.getDate()));
+            assertTrue(eventDate.isWorkingDay() || !eventDate.isWorkingDay());
```
Comment (serdaroguzhanakin) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/calendar/BankHolidaysTest.java:29]: this is always true `eventDate.isWorkingDay() || !eventDate.isWorkingDay()` innit, what is the purpose?

**Pull request closed without merging**